### PR TITLE
Mobilenet training pipeline done :)

### PR DESCRIPTION
**Pull request by DevBM04**

### CODE REVIEW

File: tamatar-ml/models/train_mobilenet.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        loss.backward()
+        optimizer.step()
+
+        running_loss += loss.item()
+
+    print(f"Epoch {epoch+1}/{epochs}, Loss: {running_loss/len(loader):.4f}")
+
+torch.save(model.state_dict(), "mobilenet.pth")
+print("Model saved!")
```
Comment (chimnayajith) [inline tamatar-ml/models/train_mobilenet.py]: Consider saving the model in a new directory. 

Also save the class names `dataset.classes` alongside the model weights to make inference easier.

File: tamatar-ml/models/train_mobilenet.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    transforms.Resize((224, 224)),
+    transforms.ToTensor(),
+    transforms.Normalize(   
+        mean=[0.485, 0.456, 0.406],
+        std=[0.229, 0.224, 0.225]
+    )
+])
+
+dataset = ImageFolder("../dataset", transform=transform)
+loader = DataLoader(dataset, batch_size=8, shuffle=True)
```
Comment (chimnayajith) [inline tamatar-ml/models/train_mobilenet.py]: utils/dataset_loader.py provides a function to load the dataset. Import and use that function for consistency.

Comment (chimnayajith) [inline tamatar-ml/models/train_mobilenet.py]: The dataset is not split into training, validation and testing sets. Consider using `torch.utils.data.random_split` to create these splits. This is also needed for the model evaluation during validation and testing.

File: tamatar-ml/models/train_mobilenet.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+num_classes = len(dataset.classes)
+model.classifier[1] = torch.nn.Linear(model.last_channel, num_classes)
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+model = model.to(device)
+
+criterion = torch.nn.CrossEntropyLoss()
+optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
+
+epochs = 5
```
Comment (chimnayajith) [inline tamatar-ml/models/train_mobilenet.py]: The number of epochs is currently hardcoded, Consider making it configurable via a command line argument or a configuration file.

**Review by chimnayajith — COMMENTED** [review]
@DevBM04 Couple of commets below. Apart from that 

- Link the issue for the mobileNet so that its automaticaly closes. Refer [this](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue)
- The script also lacks and evaluation step. Add a validation loop to calculate metrics like accuracy, precision, recall and F1 score on the validation set. This can be included in the PR decription for future reference
